### PR TITLE
Add memory cases: memory should align with 256

### PR DESCRIPTION
**Pull request by chloerh**
- cold-plug memory device
- hot-plug memory device

Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/libvirt_mem.py [hunk 1]
```diff
@@ -348,6 +389,11 @@ def modify_domain_xml():
     if not libvirt_version.version_compare(1, 2, 14):
         test.cancel("Memory hotplug not supported in current libvirt version.")
 
+    if 'align_256m' in params.get('name', ''):
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_mem.py]: I could not see the param `name` defined in config file.

File: libvirt/tests/src/libvirt_mem.py [hunk 2]
```diff
@@ -417,13 +463,29 @@ def modify_domain_xml():
                                   flagstr=max_mem_option)
             libvirt.check_exit_status(ret, maxmem_error)
 
+        # Hotplug memory device
+        if add_mem_device and hot_plug:
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_mem.py]: we do the same earlier in L377, how does the change in L423 is different.
can we have the logic squashed together avoiding redundancy

File: libvirt/tests/src/libvirt_mem.py [hunk 3]
```diff
@@ -177,6 +180,38 @@ def check_dom_xml(at_mem=False, dt_mem=False):
...
+            logging.info('%-20s:%15d', key, dom_mem[key])
+
+        if dom_mem['memory'] == sum_numa_mem + dom_mem['attached_mem']:
+            logging.info('Check Pass: Memory is equal to (all numa memory + memory device)')
+        else:
+            test.fail('Memory is not equal to (all numa memory + memory device)')
+
+        for key in dom_mem:
+            if dom_mem[key] % 256:
+                test.fail('%s not align to 256' % key)
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_mem.py]: this can be moved to L202

File: libvirt/tests/src/libvirt_mem.py (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/src/libvirt_mem.py]: I tried, but didn't find a good way without affecting existing cases.

**Comment by iccaszhulili** [comment]
rh-caselink retest please

File: libvirt/tests/src/libvirt_mem.py (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/src/libvirt_mem.py]: I don't understand, is it ok or not?

Comment (chloerh) [reply libvirt/tests/src/libvirt_mem.py]: I mean current state...

File: libvirt/tests/src/libvirt_mem.py [hunk 4]
```diff
@@ -95,7 +96,7 @@ def restore_hugepages(page_size=4):
         config.restore()
         utils_libvirtd.libvirtd_restart()
 
-    def check_qemu_cmd():
+    def check_qemu_cmd(max_mem_rt, tg_size):
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_mem.py]: miss doc string

Comment (chloerh) [reply libvirt/tests/src/libvirt_mem.py]: Updated

**Pull request merged**